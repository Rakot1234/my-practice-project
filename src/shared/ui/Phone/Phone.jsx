import React, { PureComponent } from 'react';
import './Phone.scss';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Link from '../../ui/Link/Link';

class Phone extends PureComponent {
    static propTypes = {
        phone: PropTypes.string.isRequired,
        description: PropTypes.string,
        className: PropTypes.string,
        view: PropTypes.oneOf(['header', 'footer']),
        onClickDescription: PropTypes.func
    };

    static defaultProps = {
        className: ' ',
        view: 'header'
    };

    render() {
        const {
            phone,
            description,
            className,
            view,
            onClickDescription
        } = this.props;

        return (
            <div className={cx('phone', className, `phone__view-${view}`)}>
                <Link
                    href={`tel:${phone.replace(/[^\d]/g, '')}`}
                    title={phone}
                    className={cx('phone__value')}
                />
                <div
                    className={cx('phone__description', {'phone__description_with-event': onClickDescription})}
                    onClick={onClickDescription}
                >
                    {description}
                </div>
            </div>
        );
    }
};

export default Phone;
