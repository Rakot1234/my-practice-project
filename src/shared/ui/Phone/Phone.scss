@import '../../../styles/variables.scss';

$block: phone;

.#{$block} {
    &__value {
        display: inline-block;
        color: #{$black};
        font-size: 24px;
        margin-bottom: 5px;
        @include Ubuntu;
        &:hover {
            color: #{$black};
        }
    }
    &__description {
        display: inline-block;
        cursor: pointer;
        font-size: 12px;
        color: #{$gray};
        &_with-event {
            color: #{$sandy};
            font-size: 14px;
            border-bottom: 1px dashed;
            transition: all 0.3s;
            &:hover {
                border-bottom: none;
                opacity: 0.7;
            }
        }
    }
};
