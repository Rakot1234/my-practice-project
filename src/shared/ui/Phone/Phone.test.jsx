import React from 'react';
import { shallow } from 'enzyme';
import Phone from './Phone';

describe('<Phone />', () => {
    it('Render phone field', () => {
        const wrapper = shallow(<Phone phone="8-111-11-11" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('Initiate event on click', () => {
        const onClick = jest.fn();
        const wrapper = shallow(<Phone phone="8-111-11-11" onClickDescription={onClick} />);
        wrapper.find('.phone__description').simulate('click');
        expect(onClick.mock.calls).toHaveLength(1);
    });
});
