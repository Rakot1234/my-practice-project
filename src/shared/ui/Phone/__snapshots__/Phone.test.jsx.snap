// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Phone /> Render phone field 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Phone
    phone="8-111-11-11"
    view="header"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="phone__value"
          hoverColor="yellow"
          href="tel:81111111"
          title="8-111-11-11"
        />,
        <div
          className="phone__description"
        />,
      ],
      "className": "phone phone_view_header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "phone__value",
          "hoverColor": "yellow",
          "href": "tel:81111111",
          "title": "8-111-11-11",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "phone__description",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="phone__value"
            hoverColor="yellow"
            href="tel:81111111"
            title="8-111-11-11"
          />,
          <div
            className="phone__description"
          />,
        ],
        "className": "phone phone_view_header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "phone__value",
            "hoverColor": "yellow",
            "href": "tel:81111111",
            "title": "8-111-11-11",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "phone__description",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
